### ISSUE TITLE
Parameterized URL parameters are being stripped when changing URL 

### ISSUE THREAD

**Issue by akmalnawfer** [gatsbyjs/gatsby#23983]
<!--
  I've released a website in S3, I am using query parameters to load some data, but when URL with parameters entered in the URL bar it strips the parameters and loads the default page.

Is it something to do with S3 or Gatsby.

I released the same app on a shared server its working fine.

 website link:
  - http://betterpride.com.au/
Link with parameters
 - https://betterpride.com.au/mediation?prm=dynamic

When navigated to home page with parameterized URL its working fine, but unable to land on the website with parameterized inner page URLs as I have provided above. 

I am not sure whether it is something to do with AWS S3 or Gatsby

**Comment by akmalnawfer** [comment]
<!--
  I've released a website in S3, I am using query parameters to load some data, but when URL with parameters entered in the URL bar it strips the parameters and loads the default page.

Is it something to do with S3 or Gatsby.

I released the same app on a shared server its working fine.

 website link:
  - http://betterpride.com.au/
Link with parameters
 - https://betterpride.com.au/mediation?prm=dynamic

When navigated to home page with parameterized URL its working fine, but unable to land on the website with parameterized inner page URLs as I have provided above. 

I am not sure whether it is something to do with AWS S3 or Gatsby
